sys/src/9/amd64/qmalloc.c

```diff
@@ -665,19 +655,22 @@ mallocinit(void)
 	print("base %#p ptr %#p nunits %u\n", tailbase, tailptr, tailnunits);
 }
 
-static int
-morecore(uint nunits)
-{
-	/*
-	 * First (simple) cut.
-	 * Pump it up when you don't really need it.
-	 * Pump it up until you can feel it.
-	 */
-	if(nunits < NUNITS(128*KiB))
-		nunits = NUNITS(128*KiB);
-	if(nunits > tailnunits)
-		nunits = tailnunits;
-	tailnunits -= nunits;
+static struct Allocator _qmallocAllocator = {
```

Comment:
The main reason was to separate out the core allocator code.  Since then I've been writing code to record all the allocator events so that I can analyze them and play them back in tests and benchmarks.  Breaking up the code a bit seemed a good way to enable this code to be reused when looking at implementations.

Also, after looking at the differences between the amd64 and riscv implementations, I noticed that riscv uses qmallocalign for standard qmalloc.  It seemed that with the Allocator struct I could move all the allocator code into port, and leave the Allocator definition in amd64 and riscv, with the riscv definition having qmalloc and qmallocalign pointing at the same function.

Looking at it again, something similar could be achieved without the indirection by just calling the functions in qmalloc (or some other linked implementation) directly, although not quite sure how best to handle the riscv case without duplicating code, and avoiding using the preprocessor.